humility-cmd/src/hiffy.rs

```diff
@@ -521,6 +521,27 @@ impl<'a> HiffyContext<'a> {
                         let hubris = self.hubris;
                         let f =
                             hubris.printfmt(&buf, self.failure.goff, fmt)?;
+
+                        // If Hiffy reports `Invalid`, this could be due to a
+                        // micro patch mismatch, i.e. Humility trying to use HIF
+                        // operations that the target does not know about.
+                        if f == "Some(Invalid)" {
+                            let patch = Self::read_word(
+                                hubris,
+                                core,
+                                "HIFFY_VERSION_PATCH",
+                            );
+                            match patch {
+                                Ok(patch) => if patch != HIF_VERSION_PATCH {
+                                    bail!("request failed: {0}. HIF version mismatch ({1}.{2}.{3} on host, {1}.{2}.{4} on device)",
```

Comment:
Worth phrasing as a question and then using string continuations to fit within 80 cols? e.g.:

```diff
diff --git a/humility-cmd/src/hiffy.rs b/humility-cmd/src/hiffy.rs
index 609eea9..6d5f372 100644
--- a/humility-cmd/src/hiffy.rs
+++ b/humility-cmd/src/hiffy.rs
@@ -532,14 +532,27 @@ impl<'a> HiffyContext<'a> {
                                 "HIFFY_VERSION_PATCH",
                             );
                             match patch {
-                                Ok(patch) => if patch != HIF_VERSION_PATCH {
-                                    bail!("request failed: {0}. HIF version mismatch ({1}.{2}.{3} on host, {1}.{2}.{4} on device)",
-                                        f, HIF_VERSION_MAJOR, HIF_VERSION_MINOR,
-                                        HIF_VERSION_PATCH, patch);
-                                },
-                                Err(e) =>
-                                    bail!("request failed: {}; failed to read HIF patch version: {:?}",
-                                          f, e),
+                                Ok(patch) => {
+                                    if patch != HIF_VERSION_PATCH {
+                                        bail!(
+                                            "request failed: {0}. Perhaps due \
+                                            to HIF version mismatch? \
+                                            ({1}.{2}.{3} on host, \
+                                            {1}.{2}.{4} on device)",
+                                            f,
+                                            HIF_VERSION_MAJOR,
+                                            HIF_VERSION_MINOR,
+                                            HIF_VERSION_PATCH,
+                                            patch
+                                        );
+                                    }
+                                }
+                                Err(e) => bail!(
+                                    "request failed: {}; failed to read HIF \
+                                    patch version: {:?}",
+                                    f,
+                                    e
+                                ),
                             }
                         }
                         bail!("request failed: {}", f);
```

Reply:
(And if the vertical space if offensive to the eye, could be easily refactored to be more pleasing.)

Reply:
Yes, good idea – the rightward creep is a bit of an issue here, but breaking up the line is _just okay enough_ that I don't feel obliged to fix it further.